miniconf.py

```diff
@@ -32,27 +32,53 @@ def __init__(self, client, prefix):
             client: A connected MQTT5 client.
             prefix: The MQTT toptic prefix of the device to control.
         """
+        self.uuid = uuid.uuid1()
+        self.request_id = 0
         self.client = client
         self.prefix = prefix
         self.inflight = {}
         self.client.on_message = self._handle_response
         self.client.subscribe(f'{prefix}/response/#')
+        self.logger = logging.getLogger(__name__)
 
-    def _handle_response(self, _client, topic, payload, *_args, **_kwargs):
+    def _handle_response(self, _client, _topic, payload, _qos, properties):
         """Callback function for when messages are received over MQTT.
 
         Args:
             _client: The MQTT client.
-            topic: The topic that the message was received on.
+            _topic: The topic that the message was received on.
             payload: The payload of the message.
+            _qos: The quality-of-service level of the received packet
+            properties: A dictionary of properties associated with the message.
         """
-        if topic not in self.inflight:
-            # TODO use correlation_data to distinguish clients and requests
-            logger.warning('Unexpected response on topic: %s', topic)
+        # Extract corrleation data from the properties
+        try:
+            correlation_data = json.loads(properties['correlation_data'][0].decode('ascii'))
+        except (json.decoder.JSONDecodeError, KeyError):
+            self.logger.warning('Ignoring message with invalid correlation data')
+            return
+
+        # Validate the correlation data.
+        try:
+            if correlation_data['id'] != self.uuid.hex:
+                self.logger.info('Ignoring correlation data for different ID')
+                return
+            pid = correlation_data['pid']
+        except KeyError:
+            self.logger.warning('Ignoring unknown correlation data: %s', correlation_data)
+            return
+
+        if pid not in self.inflight:
```

Comment:
I'm actually in favor of not checking this and instead taking the `KeyError` that'll be thrown automatically.
Just let the error bubble up.